pkg/agent/client.go

```diff
@@ -395,18 +389,34 @@ func (a *Client) Serve() {
 
 					close(dataCh)
 					a.connManager.Delete(connID)
-				},
-			}
-			a.connManager.Add(connID, ctx)
+				}
+				a.connManager.Add(connID, ctx)
 
-			resp.GetDialResponse().ConnectID = connID
-			if err := a.Send(resp); err != nil {
-				klog.ErrorS(err, "stream send failure")
-				continue
-			}
+				start := time.Now()
+				conn, err := net.DialTimeout(dialReq.Protocol, dialReq.Address, 5 * time.Second)
```

Comment:
nit: define a file-level constant for the timeout.

In the future we might make it a flag.

Reply:
Done with the latest commit.